src/validation.h

```diff
@@ -711,14 +732,170 @@ class SnapshotMetadata
 
 };
 
+/**
+ * Provides an interface for creating and interacting with multiple
```

Comment:
Can you give a little overview of what is meant by the:

- unready snapshot chainstate
- ibd chainstate
- snapshot chainstate
- active chainstate
- background validation chainstate
- validated chainstate

I am getting more than a little lost in the terminology, and it would nice to have it defined briefly in one place.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15606#discussion_r272327897

> Can you give a little overview of what is meant by the:
>
> - unready snapshot chainstate
> - ibd chainstate
> - snapshot chainstate
> - active chainstate
> - background validation chainstate
> - validated chainstate

More chainstates from commit "validation: introduce ChainstateManager::GetChainstateForNewBlock" (6d85241c236f563715d4a982651fd1815d76dbd2):

- most-work chainstate
- validation chainstate (different from validated chainstate?)

Reply:
Most of these are already documented with data docstrings, but I've added an overview in the classdoc.